modules/core/src/passes/pick-layers-pass.ts

```diff
@@ -135,7 +121,22 @@ export default class PickLayersPass extends LayersPass {
   }
 
   protected getLayerParameters(layer: Layer, layerIndex: number, viewport: Viewport): any {
-    const pickParameters = {...layer.props.parameters};
+    const {x, y, width, height} = viewport;
+    const pickParameters = {
+      ...layer.props.parameters,
+      scissorTest: true,
+      scissor: [x, y, width, height],
+      // TODO - When used as a custom layer in older Mapbox versions, context
+      // state was dirty. Mapbox fixed that; we should test and remove the workaround.
+      // https://github.com/mapbox/mapbox-gl-js/issues/7801
+      depthMask: true,
+      depthTest: true,
+      depthRange: [0, 1],
```

Comment:
If user disables depth test in `layer.props.parameters` it should apply to picking too